pkg/virt-controller/watch/clone/clone_base.go

```diff
@@ -262,7 +276,59 @@ func (ctrl *VMCloneController) handlePVC(obj interface{}) {
 	}
 }
 
-func (ctrl *VMCloneController) handleDeleteVM(obj interface{}) {
+func (ctrl *VMCloneController) handleAddedSourceVM(obj interface{}) {
+	vm, ok := obj.(*virtv1.VirtualMachine)
+	if !ok {
+		log.Log.Reason(fmt.Errorf("unexpected obj %#v", obj)).Error("Failed to process notification")
+		return
+	}
+	vmKey, err := cache.MetaNamespaceKeyFunc(vm)
+	if err != nil {
+		log.Log.Object(vm).Reason(err).Error("cannot get vm key")
+		return
+	}
+	keys, err := ctrl.vmCloneIndexer.IndexKeys("vmSource", vmKey)
+	if err != nil {
+		log.Log.Object(vm).Reason(err).Error("cannot get clone from vmSource indexer")
+		return
+	}
+
+	for _, k := range keys {
+		ctrl.vmCloneQueue.Add(k)
+	}
+}
+
+func (ctrl *VMCloneController) handleUpdateSourceVM(oldObj, newObj interface{}) {
+	oldVM, ok := oldObj.(*virtv1.VirtualMachine)
+	if !ok {
+		log.Log.Reason(fmt.Errorf("unexpected old obj %#v", oldObj)).Error("Failed to process notification")
+		return
+	}
+	newVM, ok := newObj.(*virtv1.VirtualMachine)
+	if !ok {
+		log.Log.Reason(fmt.Errorf("unexpected new obj %#v", newObj)).Error("Failed to process notification")
+		return
+	}
+	vmKey, err := cache.MetaNamespaceKeyFunc(newVM)
+	if err != nil {
+		log.Log.Object(newVM).Reason(err).Error("cannot get vm key")
+		return
+	}
+	// we care only for updates in a vmsource volumeSnapshotStatuses
+	keys, err := ctrl.vmCloneIndexer.IndexKeys("vmSource", vmKey)
+	if err != nil {
+		log.Log.Object(newVM).Reason(err).Error("cannot get clone from vmSource indexer")
+		return
+	}
+
+	for _, k := range keys {
+		if !equality.Semantic.DeepEqual(newVM.Status.VolumeSnapshotStatuses, oldVM.Status.VolumeSnapshotStatuses) {
+			ctrl.vmCloneQueue.Add(k)
+		}
+	}
```

Comment:
In the case that the volume snapshot statuses are equal, we can exit early and not bother about the clone keys at all

```suggestion
	if equality.Semantic.DeepEqual(newVM.Status.VolumeSnapshotStatuses, oldVM.Status.VolumeSnapshotStatuses) {
		return
	}

	vmKey, err := cache.MetaNamespaceKeyFunc(newVM)
	if err != nil {
		log.Log.Object(newVM).Reason(err).Error("cannot get vm key")
		return
	}
	// we care only for updates in a vmsource volumeSnapshotStatuses
	keys, err := ctrl.vmCloneIndexer.IndexKeys("vmSource", vmKey)
	if err != nil {
		log.Log.Object(newVM).Reason(err).Error("cannot get clone from vmSource indexer")
		return
	}

	for _, k := range keys {
		ctrl.vmCloneQueue.Add(k)
	}
```